src/lib/connector.ts

```diff
@@ -1,97 +1,93 @@
-import { Signer, ethers } from "ethers";
-import { ConnectionOptions, Connection, Token } from "../interfaces.js";
+import { Signer } from "ethers";
 import { list } from "./list.js";
-import { createToken } from "./token.js";
+import { Token } from "./token.js";
 import { read, write } from "./query.js";
 import { create } from "./create.js";
 import { hash } from "./hash.js";
 import { receipt } from "./tableland-calls.js";
-import { SUPPORTED_NETWORKS, contractAddresses } from "./util.js";
+import { SUPPORTED_CHAINS, NetworkName, ChainName } from "./util.js";
+import { Connection } from "./connection.js";
 
-declare let globalThis: any;
-
-export async function getSigner(): Promise<Signer> {
-  await globalThis.ethereum.request({ method: "eth_requestAccounts" });
-  const provider = new ethers.providers.Web3Provider(globalThis.ethereum);
-  const signer = provider.getSigner();
-  return signer;
-}
-
-export async function userCreatesToken(
-  signer: Signer,
-  chainId: number
-): Promise<Token> {
-  const now = Date.now();
-  const exp = new Date(now + 10 * 60 * 60 * 1000).toISOString(); // Default to ~10 hours
-
-  return await createToken(signer, {
-    chainId: chainId,
-    expirationTime: exp,
-    uri: globalThis.document?.location.origin,
-    version: "1",
-    statement: "Official Tableland SDK",
-  });
+/**
+ * Options to control client connection with Tableland, EVM, and Gateway.
+ */
+export interface ConnectOptions {
+  // Override any required signature by using a pre-signed SIWE token.
+  token?: Token;
+  // Signer interface for signing tokens and transactions.
+  signer?: Signer;
+  // Remote gateway host for RPC API calls.
+  host?: string;
+  // String enum indicating Tableland network. Defaults to "testnet".
+  network?: NetworkName;
+  // String enum indicating target EVM chain. Defaults to "goerli",
+  // or `signer.provider.getNetwork` if available.
+  chain?: ChainName;
+  chainId?: number;
+  // Contract address to use for Tableland Tables registry. If provided,
+  // overrides defaults derived from network + chain combination.
+  contract?: string;
 }
 
-export async function connect(options: ConnectionOptions): Promise<Connection> {
-  const network = options.network ?? "goerli";
-  const host = options.host ?? "https://testnetv2.tableland.network";
-  const contract = options.contract ?? contractAddresses[network];
-
-  if (network !== "goerli" && !options.host) {
-    throw Error(
-      "Please specify a host connection (e.g. https://testnetv2.tableland.network)"
-    );
+/**
+ * Create client connection with Tableland, EVM, and Gateway.
+ * @param options Options to control client connection.
+ * @returns Promise that resolves to a Connection object.
+ */
+export async function connect(options: ConnectOptions): Promise<Connection> {
```

Comment:
Ok, so this makes it so that you can't connect without supplying a signer right away. It will then lazily check for a signer only when it needs it. This then supports a readonly client mode where you can make queries without ever signing anything.